sdk/core/core-client-rest/src/sendRequest.ts

```diff
@@ -115,14 +115,20 @@ function buildPipelineRequest(
   url: string,
   options: InternalRequestParameters = {}
 ): PipelineRequest {
-  const { body, formData } = getRequestBody(options.body, options.contentType);
+  const { body, formData } = getRequestBody(
+    options.body,
+    options.contentType ?? (options.headers?.["content-type"] as string)
```

Comment:
contentType only exist in the RLC operation options, but in HLC and Modular, we need to use requestOptions headers content-type to pass it,
```
{
    requestOptions: { headers: { "content-type": "application/json" } }
}
```
 so we should fix it to fall back to headers content-type.

Reply:
rather than repeat this logic in multiple places, we should have a helper for determining the content type

Reply:
Sorry, I will change this.